src/sdk/namespace_fs.js

```diff
@@ -849,13 +892,18 @@ class NamespaceFS {
             throw error;
         } finally {
             try {
-                if (target_file) await target_file.close(fs_account_config);
+            dbg.log0('NamespaceFS.upload_stream:', open_mode, file_path, upload_path);
+            if (typeof nb_native().fs.temp_file === 'function' && open_mode == 'wt') {
+                await target_file.linkfileat(fs_account_config, upload_path, file_path);
+            }
```

Comment:
Cut these 3 lines (+one log line) and move it up right after the fsync call in line 887.
This way if linkfileat fails, our finally will still be called and we will close the fd (to discard it).

Reply:
BTW I would leave out this `typeof nb_native().fs.temp_file === 'function'` from here,
because open_mode=wt already checked it, and is less bloated/

Reply:
not sure if someone can use wt for other reason

Reply:
and not link it? what kind of temp file can be closed without linking?

Reply:
if it was not WT than we don't need to link it and the logic is of the old _upload_stream()